compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -914,33 +923,32 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         };
 
         if issued_spans == borrow_spans {
-            borrow_spans.var_span_label(
-                &mut err,
-                format!("borrows occur due to use of {}{}", desc_place, borrow_spans.describe(),),
-                gen_borrow_kind.describe_mutability(),
-            );
+            borrow_spans.var_subdiag(&mut err, Some(gen_borrow_kind), |kind, var_span| {
+                use crate::session_diagnostics::CaptureVarCause::*;
+                match kind {
+                    Some(_) => BorrowsUsePlaceGenerator { place: desc_place, var_span },
+                    None => BorrowsUsePlaceClosure { place: desc_place, var_span },
+                }
+            });
         } else {
-            let borrow_place = &issued_borrow.borrowed_place;
-            let borrow_place_desc = self.describe_any_place(borrow_place.as_ref());
-            issued_spans.var_span_label(
-                &mut err,
-                format!(
-                    "first borrow occurs due to use of {}{}",
-                    borrow_place_desc,
-                    issued_spans.describe(),
-                ),
-                issued_borrow.kind.describe_mutability(),
-            );
+            //FIXME: (blessed casue) I suppose this demands eager translate?
```

Comment:
Why?

Reply:
I suppose it's because 2 diag in a block, I had that before, anyway it's fixed with eager now.